Library/OAuth.swift

```diff
@@ -44,8 +54,47 @@ public struct OAuth {
       let session = ASWebAuthenticationSession(
         url: url,
         callbackURLScheme: OAuth.redirectScheme
-      ) { _, _ in
-        // TODO: MBL-1159: Exchange information in callback for credentials, then login.
+      ) { url, error in
+        guard error == nil else {
+          if let authenticationError = error as? ASWebAuthenticationSessionError,
+            authenticationError.code == .canceledLogin {
+            onComplete(.cancelled)
+          } else {
+            onComplete(.failure(errorMessage: Strings.Something_went_wrong_please_try_again()))
```

Comment:
These are all errors that aren't user-solvable, so giving them a generic error message.